RLPTests/invalidRLPTest.json

```diff
@@ -82,6 +82,51 @@
 	"nonOptimalLongLengthList2": {
 		"in": "INVALID",
 		"out": "f803112233"
+	},
+
+	"emptyEncoding": {
```

Comment:
this test fails on aleth. @gumb0

Reply:
This is a test for the empty encoding, not the empty string. An RLP encoding (i.e. the result of an RLP encoder) is never empty, it always has at least one byte. So if an RLP decoder is presented with the empty sequence of bytes, it should reject it because it is not a valid encoding. This test is meant to cover this case. Perhaps we can use a different name if it causes confusion with the test for encoding the empty string.

Reply:
How to tell the difference in the tests?

Reply:
My understanding is that RLP tests of the form
```{"in" : "INVALID", "out" : <encoding>}```
are meant to test that whether the decoder rejects `<encoding>`, while RLP tests of the form
```{"in" : <string-or-list>, "out" : <encoding>}```
are meant to test whether (1) the encoder produces `<encoding>` given `<string-or-list>` and (2) the decoder produces `<string-or-list>` given `<encoding>`.

If that's the case, then the test for the empty encoding is
```{"in" : "INVALID", "out" : ""}```
as above, while the test for the empty string is
```{"in" : "", "out" : "80"}```
Does this address your question, or have I misunderstood what you were asking?

If aleth accepts the empty encoding, I would argue that it's a bug. The specification of RLP encoding prescribes that both strings and lists start with a byte that describes the type and either the length or the length of the length. So encodings are never empty. If an RLP decoder is given a purported encoding that is empty, it must reject it because it does not encode anything. Do you agree?

Reply:
Yes.
I still will have to change aleth to treat empty string in this test as empty byte input